PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,438 @@
+                self.moveItemsBy(indexHi, (self.CELL_SIZE*deltaXHi, 0))
+                hi = mid - 1              # Yes, lower the hi boundary
+                deltaXMid = ((lo-hi)//2)-1
+                self.moveItemsBy(indexMid, (self.CELL_SIZE*deltaXMid, 0))
+
+        self.stopAnimations()
+        self.window.update()
+        self.cleanUp(callEnviron)
+        return lo                         # val not found
+
+    # delete the last element of the queue, or None if empty
+    def remove(self):
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
+        found = len(self.list) > 0
+        if found:  # Record if value was found
+            self.wait(0.3)
+
+            n = self.list[len(self.list) - 1]
+
+            indexEnd = self.createIndex(len(self.list) - 1, 'front', level=-1)  # indicate priority
+            callEnviron |= set(indexEnd)
+
+            # Slide value rectangle up and off screen
+            items = (n.display_shape, n.display_val)
+            self.moveItemsOffCanvas(items, N, sleepTime=0.02)
+
+            self.moveItemsBy(indexEnd, (-self.CELL_SIZE, 0))  # move priority arrow
+
+            self.removeFront()
+
+        self.cleanUp(callEnviron)
+        return found
+
+    def removeFront(self):
+        callEnviron = self.createCallEnvironment()
+
+        # pop a Drawable from the list
+        if len(self.list) == 0:
+            self.setMessage('Array is empty!')
+            return
+        self.startAnimations()
+        n = self.list.pop()
+
+        # delete the associated display objects
+        self.canvas.delete(n.display_shape)
+        self.canvas.delete(n.display_val)
+
+        # update window
+        self.window.update()
+        self.cleanUp(callEnviron)
+        self.stopAnimations()
+
+    def fixCells(self):       # Move canvas display items to exact cell coords
+        for i, drawItem in enumerate(self.list):
+            self.canvas.coords(drawItem.display_shape, *self.cellCoords(i))
+            self.canvas.coords(drawItem.display_val, *self.cellCenter(i))
+        self.window.update()
+
+    def cleanUp(self, *args, **kwargs): # Customize clean up for sorting
+        super().cleanUp(*args, **kwargs) # Do the VisualizationApp clean up
+        self.fixCells()       # Restore cells to their coordinates in array
+
+    def makeButtons(self):
+        vcmd = (self.window.register(numericValidate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+        newSizeArrayButton = self.addOperation(
+            "New", lambda: self.clickNew(), numArguments=1, validationCmd=vcmd)
+        insertButton = self.addOperation(
+            "Insert", lambda: self.clickInsert(), numArguments=1, validationCmd=vcmd)
+        deleteButton = self.addOperation(
+            "Remove from Front", lambda: self.clickRemove(), numArguments=1)
+        peekButton = self.addOperation(
+            "Peek", lambda: self.clickPeek(), numArguments=1)
```

Comment:
The Remove and Peek operations take no arguments. They must be added with  `numArguments=0` (which is the default, so those keyword arguments could be dropped).

The label for the Remove button should just say "Remove".